### PR TITLE
bpo-28197: Add start and stop keywords to range.index() method

### PR DESCRIPTION
**Pull request by nitishch**
Added start and stop as keywords to range.index method.

I created a temporary range object on which indexing will be done. Is there a better way to decref the temporary range object?

https://bugs.python.org/issue28197

### CODE REVIEW

**Comment by ktbarrett** [comment]
@nitishch Are you willing to get this rebased and ready for review again? If not I'll open an new PR and do that.

File: Objects/rangeobject.c [hunk 1]
```diff
@@ -527,35 +527,101 @@ range_count(rangeobject *r, PyObject *ob)
...
+            }
+        }
+        if(stop){
+            if(!PyLong_CheckExact(stop)){
+                PyErr_Format(PyExc_TypeError,
+                             "stop value must be integer, not %.200s",
+                             stop->ob_type->tp_name);
+                return NULL;
+            }
+        }
```
Comment (shreyanavigyan) [inline Objects/rangeobject.c:562]: Make the code look a little bit cleaner.
```suggestion
        if(start && (!PyLong_CheckExact(start))){
            PyErr_Format(PyExc_TypeError,
                         "start value must be integer, not %.200s",
                         start->ob_type->tp_name);
            return NULL;
        }         
        if(stop && (!PyLong_CheckExact(stop))){
            PyErr_Format(PyExc_TypeError,
                         "stop value must be integer, not %.200s",
                         stop->ob_type->tp_name);
            return NULL;
        } 
```

File: Objects/rangeobject.c [hunk 2]
```diff
@@ -527,35 +527,101 @@ range_count(rangeobject *r, PyObject *ob)
...
+        if(!start){
+            start = _PyLong_Zero;
+        }
+        if(!stop){
+            stop = r->length;
+        }
+        PyObject *slice = PySlice_New(start, stop, _PyLong_One);
+        r = (rangeobject*) compute_slice(r, slice);
+        Py_DECREF(slice);
```
Comment (shreyanavigyan) [inline Objects/rangeobject.c:571]: Slice object creation maybe unsuccessful in which case the return value maybe NULL. Therefore use Py_XDECREF. 

```suggestion
        Py_XDECREF(slice);
```

**Review by shreyanavigyan — CHANGES_REQUESTED** [review]
Found some code that requires little modifications.

**Comment by iritkatriel** [comment]
Closing as this PR is abandoned and out of date (see the issue).

**Pull request closed without merging**